drivers/adc/ad400x/ad400x.h

```diff
@@ -64,16 +64,19 @@ enum ad400x_supported_dev_ids {
 struct ad400x_dev {
 	/* SPI */
 	spi_desc *spi_desc;
+	/* Register access speed */
+	uint32_t reg_access_speed;
 	/* Device Settings */
 	enum ad400x_supported_dev_ids dev_id;
 };
 
 struct ad400x_init_param {
 	/* SPI */
 	spi_init_param spi_init;
+	/* Register access speed */
+	uint32_t reg_access_speed;
```

Comment:
Isn't the access speed specific to the chip? Can it be a define?
Also you didn't set it from the project main (at least in this commit)